src/expertsystem/io/_dict.py

```diff
@@ -24,12 +32,55 @@ def from_particle(particle: Particle) -> dict:
     )
 
 
+def from_result(result: Result) -> dict:
```

Comment:
Feels like this should be part of the specific class. What was the reasoning not to do that?

Reply:
Just the way it was already implemented for `ParticleCollection`. For sure I would keep the `asdict` a **function** (just like `attr.asdict`). The question then is whether `fromdict` should be a static **method** of the class itself or a **function** as well (as it's implemented now).

Reply:
Ok, I will just drop that design discussion here, since this will take quite a while to nail down, preventing us to wrap up the release. 

So regard the following more as a comment:
In my opinion, the attr.asdict design works because of the whole attr structure. That every attr class defines fields (attributes) and their corresponding types and values. Since a normal class does not provide that a special method has to be written for every class. Then I don't see the same design is enforced here.
So either we change the class into some plain data structure, hence attr can be used. Otherwise an interface should be defined and these functions should become the implementations of that interface. I think this will also solve this circular dependency problem that appeared 
> load_default_particles had to be moved to the top import, so that the reaction module is not dependent on the io module (the type hint for Result in the io introduces a circular dependency).

Reply:
Yes, this issue is related to #325 as well I would say, so I made a note there already.

Reply:
I'm reformulating and reopening #383. @spflueger could you list the pros and cons there?

Reply:
Ok done

Reply:
Then lets move the design discussion there